Assets/SceneChangeDoor.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.SceneManagement;
+
+public class SceneChangeDoor : MonoBehaviour
+{
+    public GameObject player;
+    public string targetScene;
+    private bool playerInInteractionArea = false;
+    private InputControls input = null;
+
+    private void Awake()
+    {
+        input = new InputControls();
+        input.Enable();
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    private void OnTriggerEnter(Collider other)
+    {
+        Debug.Log($"Door interaction zone entered by {other.name}");
+        if (other.gameObject == player)
+        {
+            playerInInteractionArea = true;
+        }
+    }
+    private void OnTriggerExit(Collider other)
+    {
+        Debug.Log($"Door interaction zone exited by {other.name}");
+        if (other.gameObject == player)
+        {
+            playerInInteractionArea = false;
+
+        }
+    }
+
+#pragma warning disable IDE0051 // This is used by the PlayerInput component
```

Comment:
Can you explain this part?

Reply:
This just disables a linter warning. `OnInteract` is a magic name from the Input system that the IDE doesn't apparently understand, so the linter was accusing this method of being unused